index.js

```diff
@@ -1,76 +1,119 @@
 #!/usr/bin/env node
 
-const fs = require('fs');
-const path = require('path');
-const csv = require('fast-csv');
+import fs from 'fs';
+import path from 'path';
+import axios from 'axios';
+import csv from 'fast-csv';
 
-const yargs = require('yargs/yargs');
-const { hideBin } = require('yargs/helpers');
+import yargs from 'yargs';
+import {hideBin} from 'yargs/helpers';
+import Logger from './logger.js';
+
+const {log, logSummary, LOG_LEVELS: {INFO, ERROR}} = Logger;
+const __dirname = path.dirname(new URL(import.meta.url).pathname);
 
 const argv = yargs(hideBin(process.argv))
-    .option("jiraUrl", {
-        type: "string",
-        describe: "Jira base url"
-    })
-    .option("confUrl", {
-        type: "string",
-        describe: "Confluence base url"
-    })
-    .option("csv", {
-        type: "string",
-        describe: "CSV filename in the current folder"
-    })
-    .option("appId", {
-        type: "string",
-        describe: "AppID can be obtained from the Application Links menu in Jira (Click on pencil icon and ID is in the URL)"
-    })
-    .demandOption(["jiraUrl", "confUrl", "csv", "appId"])
-    .argv;
+  .option("jira-url", {
+    type: "string",
+    describe: "Jira base url"
+  })
+  .option("conf-url", {
+    type: "string",
+    describe: "Confluence base url"
+  })
+  .option("csv", {
+    type: "string",
+    describe: "CSV filename in the current folder"
+  })
+  .option("app-id", {
+    type: "string",
+    describe: "AppID can be obtained from the Application Links menu in Jira (Click on pencil icon and ID is in the URL)"
+  })
+  .option("jira-auth", {
+    type: "string",
+    describe: "Username and password for the Jira user with 'Link Issues' permission in the following format 'username:password'"
+  })
+  .option("dry-run", {
+    type: "boolean",
+    default: false,
+    describe: "Script will not submit HTTP requests, if this flag is enabled."
+  })
+  .demandOption(["jiraUrl", "confUrl", "csv", "appId"])
```

Comment:
I think we should add `jira-auth` as a mandatory argument and maybe fix the option names here e.g. `jiraUrl` -> `jira-url`

Reply:
Addressed. Added `jira-auth` and fixed the naming to match the arguments. Somehow, it was still managing to detect missing arguments even before the rename, must have a camel case transformer or something similar built in 😄

Reply:
Yeah I think the library converts `camelCase` automatically :open_mouth: